<center>
    <img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/Logos/organization_logo/organization_logo.png" width="300" alt="cognitiveclass.ai logo"  />
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Peer Assignment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a Db2 database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the links below to download and save the datasets (.CSV files):

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Census Data</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Public Schools</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Crime Data</a>

**NOTE:** Ensure you have downloaded the datasets using the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be stored in the database.

While it is easier to read the dataset into a Pandas dataframe and then PERSIST it into the database as we saw in Week 3 Lab 3, it results in mapping to default datatypes which may not be optimal for SQL querying. For example a long textual field may map to a CLOB instead of a VARCHAR.

Therefore, **it is highly recommended to manually load the table using the database console LOAD tool, as indicated in Week 2 Lab 1 Part II**. The only difference with that lab is that in Step 5 of the instructions you will need to click on create "(+) New Table" and specify the name of the table you want to create and then click "Next".

<img src = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/LoadingData.png">

##### Now open the Db2 console, open the LOAD tool, Select / Drag the .CSV file for the first dataset, Next create a New Table, and then follow the steps on-screen instructions to load the data. Name the new tables as follows:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


### Connect to the database

Let us first load the SQL extension and establish a connection with the database


In [ ]:
!pip install --force-reinstall ibm_db ibm_db_sa
!pip install sqlalchemy==1.3.9


In [3]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In the next cell enter your db2 connection string. Recall you created Service Credentials for your Db2 instance in first lab in Week 3. From your Db2 service credentials copy everything after db2:// (except the double quote at the end) and paste it in the cell below after ibm_db_sa://

<img src ="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/details.png">


In [4]:
# Remember the connection string is of the format:
# %sql ibm_db_sa://my-username:my-password@my-hostname:my-port/my-db-name?security=SSL
# Enter the connection string for your Db2 on Cloud database instance below
%sql ibm_db_sa://fjy12384:jUXoAucK9zgc6UKN@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb?security=SSL

## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [6]:
%sql select * from CHICAGOCRIME limit 3

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[(3512276, 'HK587712', datetime.date(2004, 8, 28), '047XX S KEDZIE AVE', '890', 'THEFT', 'FROM BUILDING', 'SMALL RETAIL STORE', 'FALSE', 'FALSE', 911, 9, 14, 58, '6', 1155838, 1873050, 2004, Decimal('41.80744050'), Decimal('-87.70395585'), '(41.8074405, -87.703955849)'),
 (3406613, 'HK456306', datetime.date(2004, 6, 26), '009XX N CENTRAL PARK AVE', '820', 'THEFT', '$500 AND UNDER', 'OTHER', 'FALSE', 'FALSE', 1112, 11, 27, 23, '6', 1152206, 1906127, 2004, Decimal('41.89827996'), Decimal('-87.71640551'), '(41.898279962, -87.716405505)'),
 (8002131, 'HT233595', datetime.date(2011, 4, 4), '043XX S WABASH AVE', '820', 'THEFT', '$500 AND UNDER', 'NURSING HOME/RETIREMENT HOME', 'FALSE', 'FALSE', 221, 2, 3, 38, '6', 1177436, 1876313, 2011, Decimal('41.81593313'), Decimal('-87.62464213'), '(41.815933131, -87.624642127)')]

In [13]:
%sql select * from CHICAGOCENSUS limit 3

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[(1, 'Rogers Park', Decimal('7.7'), Decimal('23.6'), Decimal('8.7'), Decimal('18.2'), Decimal('27.5'), 23939, 39),
 (2, 'West Ridge', Decimal('7.8'), Decimal('17.2'), Decimal('8.8'), Decimal('20.8'), Decimal('38.5'), 23040, 46),
 (3, 'Uptown', Decimal('3.8'), Decimal('24.0'), Decimal('8.9'), Decimal('11.8'), Decimal('22.2'), 35787, 20)]

In [12]:
%sql select DISTINCT(primary_type) from CHICAGOCRIME

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[('ARSON',),
 ('ASSAULT',),
 ('BATTERY',),
 ('BURGLARY',),
 ('CONCEALED CARRY',),
 ('CRIM SEXUAL ASS',),
 ('CRIMINAL DAMAGE',),
 ('CRIMINAL TRESPA',),
 ('DECEPTIVE PRACT',),
 ('DOMESTIC VIOLEN',),
 ('GAMBLING',),
 ('HOMICIDE',),
 ('HUMAN TRAFFICKI',),
 ('INTERFERENCE WI',),
 ('INTIMIDATION',),
 ('KIDNAPPING',),
 ('LIQUOR LAW VIOL',),
 ('MOTOR VEHICLE T',),
 ('NARCOTICS',),
 ('NON - CRIMINAL',),
 ('NON-CRIMINAL',),
 ('NON-CRIMINAL (S',),
 ('OBSCENITY',),
 ('OFFENSE INVOLVI',),
 ('OTHER NARCOTIC ',),
 ('OTHER OFFENSE',),
 ('PROSTITUTION',),
 ('PUBLIC INDECENC',),
 ('PUBLIC PEACE VI',),
 ('RITUALISM',),
 ('ROBBERY',),
 ('SEX OFFENSE',),
 ('STALKING',),
 ('THEFT',),
 ('WEAPONS VIOLATI',)]

### Problem 2

##### List community areas with per capita income less than 11000.


In [15]:
%sql select com_area_name from CHICAGOCENSUS where per_capita_income < 11000

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[('West Garfield Park',),
 ('South Lawndale',),
 ('Fuller Park',),
 ('Riverdale',)]

### Problem 3

##### List all case numbers for crimes  involving minors?(children are not considered minors for the purposes of crime analysis)


In [17]:
%sql select case_number from CHICAGOCRIME 

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[('HK587712',),
 ('HK456306',),
 ('HT233595',),
 ('HT133522',),
 ('HZ138551',),
 ('HS540106',),
 ('HZ534771',),
 ('HL793243',),
 ('HL149610',),
 ('HK361551',),
 ('HX552797',),
 ('HT180228',),
 ('HV243612',),
 ('HN200365',),
 ('HK108680',),
 ('HY396729',),
 ('HR271603',),
 ('HX332132',),
 ('HW303994',),
 ('HS569246',),
 ('HY234744',),
 ('HP320451',),
 ('HN484464',),
 ('JB151690',),
 ('HR582990',),
 ('HR535632',),
 ('HH486939',),
 ('HW441620',),
 ('HJ561433',),
 ('HR583513',),
 ('HX509438',),
 ('HS681158',),
 ('HK350670',),
 ('G256545',),
 ('JA272438',),
 ('HH201585',),
 ('G212935',),
 ('HL590300',),
 ('HL409680',),
 ('HM324025',),
 ('G662598',),
 ('HV353560',),
 ('HR261581',),
 ('HL680775',),
 ('HM543491',),
 ('HY167451',),
 ('HY377901',),
 ('HX225704',),
 ('HJ308632',),
 ('HV528008',),
 ('HK801112',),
 ('HS532107',),
 ('JA261898',),
 ('JA559176',),
 ('JA412895',),
 ('HY273740',),
 ('HZ224258',),
 ('HJ652747',),
 ('G040244',),
 ('HJ415128',),
 ('HV534721',),
 ('JA503725',),
 ('HM803061',),
 ('HN731220',),
 ('HP691879',),
 ('HV537004',),
 ('HR149903',),
 ('HX314361',),
 ('HV345270',),
 ('HS308747',),
 ('HK728309',),
 ('HL005270',),
 ('HY291329',),
 ('HJ678713',),
 ('HY412959',),
 ('HX155070',),
 ('HV402967',),
 ('HX332979',),
 ('HJ710634',),
 ('HX165034',),
 ('HR385117',),
 ('HS230912',),
 ('HL177398',),
 ('HH702524',),
 ('JA384163',),
 ('HJ672058',),
 ('HT368471',),
 ('HY202462',),
 ('HX354359',),
 ('HZ146741',),
 ('G376790',),
 ('HX262353',),
 ('HP317516',),
 ('HZ335434',),
 ('HS678732',),
 ('HL382289',),
 ('HM392612',),
 ('HH869886',),
 ('HR460350',),
 ('HM117889',),
 ('HR576816',),
 ('HZ372381',),
 ('HZ556726',),
 ('HN421595',),
 ('HK454598',),
 ('HH657809',),
 ('HS451246',),
 ('HH316735',),
 ('G597730',),
 ('G736811',),
 ('HT651635',),
 ('HZ432273',),
 ('G517474',),
 ('HP746347',),
 ('G056330',),
 ('HN772519',),
 ('HZ309936',),
 ('HY368284',),
 ('HL353697',),
 ('HT375934',),
 ('JA188596',),
 ('HL725506',),
 ('HV395700',),
 ('HH696901',),
 ('HT390647',),
 ('HK815196',),
 ('JA490957',),
 ('HK103473',),
 ('HM556695',),
 ('HH558381',),
 ('HK498590',),
 ('HX455839',),
 ('HR703960',),
 ('HL646682',),
 ('HV445580',),
 ('G688026',),
 ('HJ466766',),
 ('HL632721',),
 ('HP579910',),
 ('HL718742',),
 ('HP714065',),
 ('HP716225',),
 ('HW481295',),
 ('HS568476',),
 ('JB213238',),
 ('HS475732',),
 ('HS595785',),
 ('HR408597',),
 ('HT542713',),
 ('HL240985',),
 ('HJ254782',),
 ('HK561262',),
 ('HW213573',),
 ('G419948',),
 ('HL177008',),
 ('HJ639869',),
 ('HM217017',),
 ('HL388623',),
 ('HY189363',),
 ('HS498285',),
 ('HN224900',),
 ('HR387668',),
 ('HJ294892',),
 ('HL745027',),
 ('G533278',),
 ('HH827769',),
 ('HH639427',),
 ('HK771992',),
 ('HH867007',),
 ('HJ575773',),
 ('HR136920',),
 ('HT457930',),
 ('G070193',),
 ('HJ724585',),
 ('HP121474',),
 ('G198370',),
 ('HJ723825',),
 ('HJ458342',),
 ('HP558812',),
 ('HY192694',),
 ('HN472326',),
 ('HK756778',),
 ('HH741961',),
 ('JA460432',),
 ('G140454',),
 ('HX239866',),
 ('HM378171',),
 ('HW182288',),
 ('HZ295843',),
 ('HM673431',),
 ('HP352546',),
 ('HR286405',),
 ('HR195987',),
 ('HW261668',),
 ('JA496716',),
 ('JA344492',),
 ('HZ334220',),
 ('HR217367',),
 ('HV268964',),
 ('JA483294',),
 ('HT169840',),
 ('HT621306',),
 ('HS269736',),
 ('HK243330',),
 ('HH262924',),
 ('HL613433',),
 ('HY267896',),
 ('HN568531',),
 ('HT112422',),
 ('HV479133',),
 ('HV592346',),
 ('HH619797',),
 ('HZ376709',),
 ('HV573337',),
 ('HL269210',),
 ('HL567458',),
 ('HR674429',),
 ('HM200888',),
 ('HX467661',),
 ('HX448587',),
 ('HS200939',),
 ('HN335611',),
 ('HZ535494',),
 ('HX154767',),
 ('HM337445',),
 ('HM244120',),
 ('HR342598',),
 ('HL187666',),
 ('HW449676',),
 ('HS589302',),
 ('HX145326',),
 ('HN553354',),
 ('HN784049',),
 ('HJ527694',),
 ('HM303461',),
 ('HM535632',),
 ('HT353855',),
 ('G134016',),
 ('HR125162',),
 ('G496261',),
 ('HJ842548',),
 ('HN280980',),
 ('JA560123',),
 ('HY405637',),
 ('HP689703',),
 ('HR345366',),
 ('HZ242802',),
 ('HM661326',),
 ('HJ336995',),
 ('HK694552',),
 ('HJ760939',),
 

### Problem 4

##### List all kidnapping crimes involving a child?


In [23]:
%sql select * from CHICAGOCRIME where primary_type = 'KIDNAPPING'

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[(5276766, 'HN144152', datetime.date(2007, 1, 26), '050XX W VAN BUREN ST', '1792', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER', 'STREET', 'FALSE', 'FALSE', 1533, 15, 29, 25, '20', 1143050, 1897546, 2007, Decimal('41.87490841'), Decimal('-87.75024931'), '(41.874908413, -87.750249307)')]

### Problem 5

##### What kinds of crimes were recorded at schools?


In [27]:
# %sql select UNIQUE(location_description) from CHICAGOCRIME

In [32]:
%sql select location_description from CHICAGOCRIME where location_description like SCHOOL%

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
(ibm_db_dbi.ProgrammingError) ibm_db_dbi::ProgrammingError: SQLNumResultCols failed: [IBM][CLI Driver][DB2/LINUXX8664] SQL0104N  An unexpected token "END-OF-STATEMENT" was found following "ription like SCHOOL%".  Expected tokens may include:  "<factor>".  SQLSTATE=42601\r SQLCODE=-104
[SQL: select location_description from CHICAGOCRIME where location_description like SCHOOL%]
(Background on this error at: http://sqlalche.me/e/f405)


### Problem 6

##### List the average safety score for all types of schools.


In [34]:
%sql select * from CHICAGOSCH where (select avg(safety_score) from CHICAGOSCH)

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[(610038, 'Abraham Lincoln Elementary School', 'ES', '615 W Kemper Pl', 'Chicago', 'IL', 60614, '(773) 534-5720', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610038.pdf', 'Fullerton Elementary Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 1', 'Yes', 'Very Strong', 99, 'Very Strong', '99', 'Strong', 74, 'Strong', 66, 'Weak', '65', 'Strong', '70', 'Strong', '56', 'Average', '47', '96.00%', Decimal('2.0'), '96.40%', '95.80%', '80.1', '43.3', '89.6', '84.9', '60.7', '62.6', '81.9', '85.2', '52', '62.4', '66.3', '77.9', Decimal('69.7'), Decimal('64.4'), Decimal('0.2'), Decimal('0.9'), 'Yellow', 'Green', '67.1', '54.5', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 813, 33, 'NDA', Decimal('1171699.458'), Decimal('1915829.428'), Decimal('41.92449696'), Decimal('-87.64452163'), 7, 'LINCOLN PARK', 43, 18, '(41.92449696, -87.64452163)'),
 (610281, 'Adam Clayton Powell Paideia Community Academy Elementary School', 'ES', '7511 S South Shore Dr', 'Chicago', 'IL', 60649, '(773) 535-6650', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610281.pdf', 'Skyway Elementary Network', 'SOUTH SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 54, 'Strong', '66', 'Strong', 74, 'Very Strong', 84, 'Weak', '63', 'Strong', '76', 'Weak', '46', 'Average', '50', '95.60%', Decimal('15.7'), '95.30%', '100.00%', '62.4', '51.7', '21.9', '15.1', '29', '42.8', '38.5', '27.4', '44.8', '42.7', '14.1', '34.4', Decimal('16.8'), Decimal('16.5'), Decimal('0.7'), Decimal('1.4'), 'Green', 'Green', '17.2', '27.3', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 521, 46, 'NDA', Decimal('1196129.985'), Decimal('1856209.466'), Decimal('41.76032435'), Decimal('-87.55673627'), 43, 'SOUTH SHORE', 7, 4, '(41.76032435, -87.55673627)'),
 (610185, 'Adlai E Stevenson Elementary School', 'ES', '8010 S Kostner Ave', 'Chicago', 'IL', 60652, '(773) 535-2280', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610185.pdf', 'Midway Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 2', 'No', 'Strong', 61, 'NDA', 'NDA', 'Average', 50, 'Weak', 36, 'Weak', 'NDA', 'NDA', 'NDA', 'Average', '47', 'Weak', '41', '95.70%', Decimal('2.3'), '94.70%', '98.30%', '53.7', '26.6', '38.3', '34.7', '43.7', '57.3', '48.8', '39.2', '46.8', '44', '7.5', '21.9', Decimal('18.3'), Decimal('15.5'), Decimal('-0.9'), Decimal('-1.0'), 'Red', 'Red', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 1324, 44, 'NDA', Decimal('1148427.165'), Decimal('1851012.215'), Decimal('41.74711093'), Decimal('-87.73170248'), 70, 'ASHBURN', 13, 8, '(41.74711093, -87.73170248)'),
 (609993, 'Agustin Lara Elementary Academy', 'ES', '4619 S Wolcott Ave', 'Chicago', 'IL', 60609, '(773) 535-4389', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_609993.pdf', 'Pershing Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 56, 'Average', '44', 'Average', 45, 'Weak', 37, 'Weak', '65', 'Average', '48', 'Average', '53', 'Strong', '58', '95.50%', Decimal('10.4'), '95.80%', '100.00%', '76.9', 'NDA', '26', '24.7', '61.8', '49.7', '39.2', '27.2', '69.7', '60.6', '9.1', '18.2', Decimal('11.1'), Decimal('9.6'), Decimal('0.9'), Decimal('2.4'), 'Green', 'Green', '42.9', '25', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 556, 42, 'NDA', Decimal('1164504.290'), Decimal('1873959.199'), Decimal('41.80975690'), Decimal('-87.67214460'), 61, 'NEW CITY', 20, 9, '(41.8097569, -87.6721446)'),
 (610513, 'Air Force Academy High School', 'HS', '3630 S Wells St', 'Chicago', 'IL', 60609, '(773) 535-1590', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610513.pdf', 'Southwest Side High School Network', 'SOUTHWEST SIDE COLLABORATIVE', 'NDA', 'Standard', 'Not on Probation'

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [40]:
%sql select com_area_name from CHICAGOCENSUS where (select min(percent_households_below_poverty) from CHICAGOCENSUS) 

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
(ibm_db_dbi.ProgrammingError) ibm_db_dbi::ProgrammingError: SQLNumResultCols failed: [IBM][CLI Driver][DB2/LINUXX8664] SQL0401N  The data types of the operands for the operation "=" are not compatible or comparable.  SQLSTATE=42818\r SQLCODE=-401
[SQL: select com_area_name from CHICAGOCENSUS where (select min(percent_households_below_poverty) from CHICAGOCENSUS)]
(Background on this error at: http://sqlalche.me/e/f405)


### Problem 8

##### Which community area is most crime prone?


In [45]:
%sql select community_area_number from CHICAGOCRIME

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[(58,),
 (23,),
 (38,),
 (46,),
 (66,),
 (24,),
 (14,),
 (38,),
 (72,),
 (21,),
 (65,),
 (28,),
 (49,),
 (28,),
 (26,),
 (22,),
 (48,),
 (44,),
 (25,),
 (25,),
 (8,),
 (32,),
 (69,),
 (59,),
 (68,),
 (14,),
 (76,),
 (66,),
 (73,),
 (44,),
 (32,),
 (35,),
 (6,),
 (None,),
 (30,),
 (None,),
 (None,),
 (23,),
 (68,),
 (8,),
 (None,),
 (43,),
 (31,),
 (24,),
 (5,),
 (68,),
 (22,),
 (32,),
 (25,),
 (6,),
 (68,),
 (42,),
 (25,),
 (29,),
 (23,),
 (8,),
 (25,),
 (28,),
 (None,),
 (59,),
 (7,),
 (67,),
 (6,),
 (32,),
 (39,),
 (77,),
 (6,),
 (31,),
 (61,),
 (49,),
 (23,),
 (25,),
 (2,),
 (63,),
 (67,),
 (69,),
 (64,),
 (62,),
 (66,),
 (75,),
 (65,),
 (49,),
 (22,),
 (28,),
 (21,),
 (24,),
 (1,),
 (15,),
 (30,),
 (71,),
 (None,),
 (76,),
 (64,),
 (16,),
 (68,),
 (8,),
 (77,),
 (25,),
 (67,),
 (50,),
 (14,),
 (28,),
 (73,),
 (56,),
 (25,),
 (51,),
 (5,),
 (None,),
 (None,),
 (None,),
 (25,),
 (21,),
 (None,),
 (2,),
 (None,),
 (71,),
 (39,),
 (30,),
 (43,),
 (68,),
 (61,),
 (4,),
 (29,),
 (43,),
 (27,),
 (24,),
 (49,),
 (21,),
 (23,),
 (25,),
 (8,),
 (50,),
 (25,),
 (73,),
 (6,),
 (None,),
 (68,),
 (31,),
 (42,),
 (23,),
 (25,),
 (35,),
 (29,),
 (30,),
 (19,),
 (19,),
 (46,),
 (27,),
 (6,),
 (7,),
 (19,),
 (29,),
 (43,),
 (None,),
 (68,),
 (37,),
 (68,),
 (26,),
 (32,),
 (6,),
 (30,),
 (67,),
 (25,),
 (69,),
 (None,),
 (25,),
 (25,),
 (29,),
 (16,),
 (28,),
 (28,),
 (23,),
 (None,),
 (29,),
 (7,),
 (None,),
 (39,),
 (8,),
 (62,),
 (66,),
 (23,),
 (61,),
 (25,),
 (70,),
 (None,),
 (27,),
 (37,),
 (38,),
 (29,),
 (49,),
 (61,),
 (25,),
 (69,),
 (26,),
 (24,),
 (54,),
 (2,),
 (25,),
 (60,),
 (67,),
 (58,),
 (10,),
 (41,),
 (52,),
 (None,),
 (24,),
 (53,),
 (61,),
 (58,),
 (6,),
 (23,),
 (31,),
 (44,),
 (29,),
 (22,),
 (66,),
 (63,),
 (10,),
 (29,),
 (50,),
 (25,),
 (15,),
 (58,),
 (4,),
 (19,),
 (1,),
 (31,),
 (24,),
 (26,),
 (51,),
 (30,),
 (68,),
 (70,),
 (16,),
 (16,),
 (11,),
 (67,),
 (None,),
 (62,),
 (None,),
 (24,),
 (5,),
 (18,),
 (63,),
 (44,),
 (24,),
 (70,),
 (17,),
 (47,),
 (43,),
 (8,),
 (48,),
 (49,),
 (71,),
 (25,),
 (22,),
 (55,),
 (44,),
 (3,),
 (25,),
 (43,),
 (27,),
 (None,),
 (1,),
 (70,),
 (58,),
 (25,),
 (76,),
 (28,),
 (26,),
 (29,),
 (3,),
 (None,),
 (69,),
 (25,),
 (68,),
 (29,),
 (28,),
 (3,),
 (46,),
 (58,),
 (63,),
 (38,),
 (27,),
 (66,),
 (71,),
 (23,),
 (61,),
 (42,),
 (42,),
 (25,),
 (61,),
 (31,),
 (26,),
 (28,),
 (23,),
 (53,),
 (3,),
 (29,),
 (23,),
 (23,),
 (23,),
 (23,),
 (15,),
 (25,),
 (25,),
 (25,),
 (69,),
 (58,),
 (66,),
 (14,),
 (22,),
 (1,),
 (23,),
 (61,),
 (62,),
 (6,),
 (31,),
 (8,),
 (19,),
 (39,),
 (42,),
 (23,),
 (15,),
 (18,),
 (21,),
 (2,),
 (23,),
 (25,),
 (67,),
 (51,),
 (46,),
 (44,),
 (69,),
 (26,),
 (45,),
 (24,),
 (68,),
 (58,),
 (25,),
 (2,),
 (8,),
 (44,),
 (25,),
 (23,),
 (25,),
 (33,),
 (25,),
 (31,),
 (51,),
 (49,),
 (27,),
 (68,),
 (6,),
 (26,),
 (None,),
 (68,),
 (72,),
 (38,),
 (45,),
 (42,),
 (63,),
 (38,),
 (71,),
 (27,),
 (8,),
 (71,),
 (68,),
 (49,),
 (43,),
 (25,),
 (None,),
 (68,),
 (28,),
 (25,),
 (49,),
 (46,),
 (29,),
 (None,),
 (19,),
 (24,),
 (None,),
 (25,),
 (53,),
 (61,),
 (None,),
 (7,),
 (65,),
 (69,),
 (52,),
 (20,),
 (69,),
 (30,),
 (72,),
 (17,),
 (17,),
 (72,),
 (48,),
 (68,),
 (43,),
 (71,),
 (67,),
 (68,),
 (25,),
 (67,),
 (68,),
 (16,),
 (46,),
 (16,),
 (28,),
 (48,),
 (41,),
 (None,),
 (31,),
 (22,),
 (42,),
 (26,),
 (28,),
 (69,),
 (None,),
 (68,),
 (66,),
 (25,),
 (63,),
 (71,),
 (19,),
 (58,),
 (66,),
 (76,),
 (8,),
 (2,),
 (None,),
 (71,),
 (24,),
 (None,),
 (8,),
 (1,),
 (70,),
 (None,),
 (24,),
 (71,),
 (16,),
 (6,),
 (25,),
 (None,),
 (20,),
 (None,),
 (46,),
 (38,),
 (2,),
 (29,),
 (30,),
 (53,),
 (23,),
 (71,),
 (5,),
 (23,),
 (67,),
 (14,),
 (15,),
 (25,),
 (10,),
 (28,),
 (69,),
 (53,),
 (44,),
 (26,),
 (49,),
 (None,),
 (35,),
 (49,),
 (70,),
 (71,),
 (66,),
 (22,),
 (32,),
 (28,),
 (35,),
 (35,),
 (38,),
 (32,),
 (4,),
 (None,),
 (None,),
 (1,),
 (None,),
 (None,),
 (8,),
 (25,),
 (25,),
 (66,),
 (29,),
 (67,),
 (22,),
 (46,),
 (63,),
 (56,)

Double-click **here** for a hint

<!--
Query for the 'community area number' that is most crime prone.
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [50]:
%sql select com_area_name from CHICAGOCRIME

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
(ibm_db_dbi.ProgrammingError) ibm_db_dbi::ProgrammingError: SQLNumResultCols failed: [IBM][CLI Driver][DB2/LINUXX8664] SQL0206N  "COM_AREA_NAME" is not valid in the context where it is used.  SQLSTATE=42703\r SQLCODE=-206
[SQL: select com_area_name from CHICAGOCRIME]
(Background on this error at: http://sqlalche.me/e/f405)


In [52]:
%sql select com_area_name from CHICAGOCENSUS where hardship_index = (select max(hardship_index) from CHICAGOCENSUS )

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [58]:
%sql select * from CHICAGOCENSUS limit 3

 * ibm_db_sa://fjy12384:***@1bbf73c5-d84a-4bb0-85b9-ab1a4348f4a4.c3n41cmd0nqnrk39u98g.databases.appdomain.cloud:32286/bludb
Done.


[(1, 'Rogers Park', Decimal('7.7'), Decimal('23.6'), Decimal('8.7'), Decimal('18.2'), Decimal('27.5'), 23939, 39),
 (2, 'West Ridge', Decimal('7.8'), Decimal('17.2'), Decimal('8.8'), Decimal('20.8'), Decimal('38.5'), 23040, 46),
 (3, 'Uptown', Decimal('3.8'), Decimal('24.0'), Decimal('8.9'), Decimal('11.8'), Decimal('22.2'), 35787, 20)]

Copyright © 2020 [cognitiveclass.ai](cognitiveclass.ai?utm_source=bducopyrightlink&utm_medium=dswb&utm_campaign=bdu). This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
| 2021-11-17 | 2.6     | Lakshmi           | Updated library                                |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
